xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -553,14 +555,17 @@ private void handleLdsResponseForConfigUpdate(DiscoveryResponse ldsResponse) {
     List<String> listenerNames = new ArrayList<>(ldsResponse.getResourcesCount());
     try {
       for (com.google.protobuf.Any res : ldsResponse.getResourcesList()) {
+        if (res.getTypeUrl().equals(ADS_TYPE_URL_LDS_v2)) {
```

Comment:
Couple of things:
- based on https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/discovery/v3/discovery.proto#L97 which says "Type URL ... Must be consistent with the type_url in the 'resources' repeated Any". Should we have that check here now that v2 and v3 are the 2 possibilities?
- shouldn't the typeUrl in ACK/NACK match what was there in the response? I think it is possible that the client asks for v3 in the request but the response comes back with v2. In that case should ACK/NACK match the typeUrl of the original request or the response?

Reply:
I did have the check that v2 and v3 are the 2 possibilities (Will ACK for 2 possibilities)
https://github.com/grpc/grpc-java/pull/7169/files/6f32fb0eec0589cd4d606418de35ca2a85cfb855#diff-2ab746208e54edd7bcbba5fc408b6946R1327

But I do not check the type_url of response is consistent with that of repeated Any - I assume the TD is consistent.

Reply:
>In that case should ACK/NACK match the typeUrl of the original request or the response?

I think we should match the response.

Reply:
> > In that case should ACK/NACK match the typeUrl of the original request or the response?
> 
> I think we should match the response.

That discussion is in a separate thread so we can continue it there. I agree that there is no need to validate responses from the control plane (TD).